Gems/Atom/RHI/DX12/Code/Source/Platform/Windows/RHI/Device_Windows.cpp

```diff
@@ -225,6 +225,9 @@ namespace AZ
             m_dxgiFactory = physicalDevice.GetFactory();
             m_dxgiAdapter = physicalDevice.GetAdapter();
 
+            DEVMODE DisplayConfig;
```

Comment:
Thinking on this, I'm not sure the idea of a "main" display is going to work out for the Editor or PC platform in general.

The Editor main window can be moved from display to display, and PC games should be multi-display aware.

One option would be tie the refresh rate to the swap chain / `NativeWindow` if we can't easily query that from DX12/Vulkan and instead need to go through the OS.

Alternatively the RHI system will could provide some kind of map of displays to refresh rates, and we could map windows to a given display ID.

On Windows, I think the usual approach is to listen for `WM_WINDOWPOSCHANGED` and call `EnumDisplayInfo` to make sure the refresh rate hasn't changed on you, but we might have better options here.

Reply:
So this is first pass so we can get something up and running quick. Ideally we should have a callback for each platform to update the refresh rate in case the display changes but that is a lot of work as every platform has different api and testing each one will require multi screen setup which I dont currently have at home. 

Also, as I was adding this I was wondering if this code needs to be in RHI at all. We are not really using driver api here and instead using OS api although that can change (maybe)? I know that vulkan has some api but its not really supported in most of the hardware so its useless. Currently all the api is OS specific for refresh rate. I was wondering if the code to manage refresh rate is better located as part of NativeWindow. It could be moved later too. Thoughts?

Reply:
I think moving this to `NativeWindow` actually does make the most sense (and afaik `NativeWindow` will always sort of be bundled with Atom as part of `AzFramework`?) - I did something similar for DPI scaling not too long ago, I think anything that's "query the OS" as opposed to "query the renderer" for a given window handle can sensibly live there.